modules/mobile/spec/request/prescriptions_request_spec.rb

```diff
@@ -0,0 +1,280 @@
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to match_json_schema('prescription')
+        end
+      end
+
+      context 'returns a 403 forbidden response' do
+        before do
+          unauthorized_user = build(:iam_user)
+          iam_sign_in(unauthorized_user)
+
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get '/mobile/v0/health/rx/prescriptions', headers: iam_headers
+          end
+        end
+
+        it 'returns expected error response', :aggregate_failures do
+          expect(response).to have_http_status(:forbidden)
+          expect(response.parsed_body).to eq({ 'errors' =>
+                                                 [{ 'title' => 'Forbidden',
+                                                    'detail' => 'User does not have access to the requested resource',
+                                                    'code' => '403',
+                                                    'status' => '403' }] })
+        end
+      end
+    end
+
+    context 'pagination parameters' do
+      context 'only valid pagination parameters' do
+        let(:params) { { page: { number: 2, size: 3 } } }
+
+        it 'paginates as expected', :aggregate_failures do
+          VCR.use_cassette('rx_refill/prescriptions/gets_a_list_of_all_prescriptions') do
+            get '/mobile/v0/health/rx/prescriptions', params: params, headers: iam_headers
+          end
+          expect(response).to have_http_status(:ok)
+          expect(response.body).to match_json_schema('prescription')
+          expect(response.parsed_body['meta']).to eq({ 'pagination' =>
+                                                        { 'currentPage' => 2,
+                                                          'perPage' => 3,
+                                                          'totalPages' => 5,
+                                                          'totalEntries' => 14 } })
+          expect(response.parsed_body['links']).to eq(
+            { 'self' =>
+              'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=2',
+              'first' =>
+              'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=1',
+              'prev' =>
+              'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=1',
+              'next' =>
+              'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=3',
+              'last' =>
+              'http://www.example.com/mobile/v0/health/rx/prescriptions?page[size]=3&page[number]=5' }
+          )
+        end
+      end
+
+      context 'valid pagination parameters with other parameters' do
+        let(:params) { { page: { number: 2, size: 3 }, sort: '-refill_date' } }
+        let(:filter_param) { 'filter[[refill_status][eq]]=refillinprocess' }
```

Comment:
I wonder if you could avoid tacking this on as a string if you changed the params like:
```
{ 'sort' => '-refill_date', 'filter[[refill_status][eq]]' => 'refillinprocess' }
```

Reply:
I'm leveraging the vets-api common::collections sort which uses that `-` to get the inverse. We'd need to build out our own sort to do that though I agree that'd be ideal.